qiskit/wrapper/defaultqiskitprovider.py

```diff
@@ -103,6 +110,16 @@ def add_provider(self, provider, provider_name):
                 'A provider with name "%s" is already registered.'
                 % provider_name)
 
+        # Check for backend name clashes, emitting a warning.
+        current_backends = {str(backend) for backend in self.available_backends()}
+        added_backends = {str(backend) for backend in provider.available_backends()}
```

Comment:
Non-blocking: can we make it clear with the name that we did not add the backends yet? That they are `incoming_backends` or `new_provider_backends` or something like this?